jsregexp.c

```diff
@@ -63,13 +64,33 @@ static int regex_closure(lua_State *lstate)
     lua_pushinteger(lstate, capture[1] - input);
     lua_setfield(lstate, -2, "end_ind");
 
-    lua_newtable(lstate);
+    lua_newtable(lstate); // groups
+    lua_newtable(lstate); // named_groups
+
+    const char* group_names = NULL;
+    if (named_groups) {
+      group_names = lre_get_groupnames(r->bc);
```

Comment:
I think we can just  push the table for named groups here so we don't have to pop it later. Need to switch the stack indexes `-2` and `-3` in the loop, of course.

Reply:
Uhh I’m not quite sure how to do this without having some sort of ternarys  to specify stack position

Reply:
I should have made this clearer. If the table of the normal indexed matches is on top, one can always access it with `rawseti(lstate, -2, ...)`. If `named_groups` is true, have the table of named groups *below* that so it is accessed with `setfield(lstate, -3, ...)`. Just make sure you push a second table when `named_groups` is true.

Reply:
I have updated it to work somewhat as you describe. I'm pretty sure you have to do some branching for stack positions as is done [here](https://github.com/kmarius/jsregexp/pull/8/files#diff-bf832cc2ca0e905b0def25d6989e2ead4b330c512a0a651094d8eaeee86cc4e8R88-R92)